orchagent/aclorch.cpp

```diff
@@ -2129,69 +2143,71 @@ void AclOrch::queryAclActionCapability()
         {
-            auto action = static_cast<sai_acl_action_type_t>(action_list[i]);
-            m_aclCapabilities[stage].insert(action);
-            SWSS_LOG_INFO("    %s", sai_serialize_enum(action, &sai_metadata_enum_sai_acl_action_type_t).c_str());
-        }
+            auto stage = (stage_attr == SAI_SWITCH_ATTR_ACL_STAGE_INGRESS ? ACL_STAGE_INGRESS : ACL_STAGE_EGRESS);
+            auto stage_str = (stage_attr == SAI_SWITCH_ATTR_ACL_STAGE_INGRESS ? STAGE_INGRESS : STAGE_EGRESS);
+            action_list.resize(static_cast<size_t>(max_action_count));
 
-        // put capabilities in state DB
+            attr.id = stage_attr;
+            attr.value.aclcapability.action_list.list  = action_list.data();
+            attr.value.aclcapability.action_list.count = max_action_count;
 
-        auto field = std::string("ACL_ACTIONS") + '|' + stage_str;
-        auto& acl_action_set = m_aclCapabilities[stage];
+            status = sai_switch_api->get_switch_attribute(gSwitchId, 1, &attr);
+            if (status == SAI_STATUS_SUCCESS)
+            {
 
-        string delimiter;
-        ostringstream acl_action_value_stream;
+                SWSS_LOG_INFO("Supported %s action count %d:", stage_str,
+                              attr.value.aclcapability.action_list.count);
 
-        for (const auto& action_map: {aclL3ActionLookup, aclMirrorStageLookup, aclDTelActionLookup})
-        {
-            for (const auto& it: action_map)
-            {
-                auto saiAction = getAclActionFromAclEntry(it.second);
-                if (acl_action_set.find(saiAction) != acl_action_set.cend())
+                for (size_t i = 0; i < static_cast<size_t>(attr.value.aclcapability.action_list.count); i++)
                 {
-                    acl_action_value_stream << delimiter << it.first;
-                    delimiter = comma;
+                    auto action = static_cast<sai_acl_action_type_t>(action_list[i]);
+                    m_aclCapabilities[stage].insert(action);
+                    SWSS_LOG_INFO("    %s", sai_serialize_enum(action, &sai_metadata_enum_sai_acl_action_type_t).c_str());
                 }
             }
-        }
+            else if (status == SAI_STATUS_NOT_IMPLEMENTED)
+            {
+                SWSS_LOG_WARN("Failed to query ACL %s action capabilities - "
+                        "API not implemented, using defaults",
+                        stage_str);
+                initDefaultAclActionCapabilities(stage);
+            }
+            else
+            {
+                SWSS_LOG_THROW("AclOrch initialization failed: "
+                               "failed to query supported %s ACL actions", stage_str);
+            }
 
-        fvVector.emplace_back(field, acl_action_value_stream.str());
-        m_switchTable.set("switch", fvVector);
+            // put capabilities in state DB
+            putAclActionCapabilityInDB(stage);
+        }
     }
+    else if (status == SAI_STATUS_NOT_IMPLEMENTED)
```

Comment:
Please have just the `else `case to populate the default table

Reply:
I changed it. But that means SAI does not return valid status code